doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1591,21 +1592,172 @@ Numeral notations
          function application, constructors, inductive type families,
          sorts, and primitive integers) will be considered for printing.
 
-      :n:`( warning after @numeral )`
+         .. note::
+            Numeral notations for parameterized inductive type can be
+            added by declaring an :ref:`abbreviation <Abbreviations>`
+            for the inductive which instantiates all parameters.
+
+            .. coqtop:: in reset
+
+               Definition of_uint (u : Numeral.uint) : list unit :=
+                 let fix f n :=
+                   match n with O => nil | S n => cons tt (f n) end in
+                 f (Nat.of_num_uint u).
+               Definition to_uint (l : list unit) : Numeral.uint :=
+                 let fix f n :=
+                   match n with nil => O | cons tt l => S (f l) end in
+                 Nat.to_num_uint (f l).
+
+               Notation listunit := (list unit) (only parsing).
+               Numeral Notation listunit of_uint to_uint : nat_scope.
+
+            .. coqtop:: all
+
+               Check 0.
+               Check 1.
+               Check 2.
+
+      :n:`via @qualid__ind using ( {+, @qualid @qualid } )`
+         When using this variant, the :n:`@qualid` type above no
+         longer needs to be an inductive. It is mapped to the
+         inductive type (without parameters) :n:`@qualid__ind` according to the provided
+         list of pairs :n:`@qualid @qualid` whose first component is a
+         constant of type :n:`@qualid` above
+         (or a function of type :n:`{* _ -> } @qualid`) and the second a
+         constructor of type :n:`@qualid__ind`. The above type
+         :n:`@qualid` is then replaced by :n:`@qualid__ind` in the
+         above parser and printer types.
+
+         .. note::
+            To use a sort as target type, use an :ref:`abbreviation <Abbreviations>`
+            as in example below.
+
+         The following example parses and prints :g:`prod unit (prod unit unit)` as :g:`3`.
```

Comment:
Would you move this example to the end of the subsection and add a hyperlink here?

We like to keep the syntax explanations of each element brief so the command syntax and the explanations (e.g. the box below and the following text) are in the same screen full as much as possible.  And easy for the reader to scan.  This command description on this point already deserved some rewriting on this point.

![image](https://user-images.githubusercontent.com/1253341/91906030-5c6cc500-ec5c-11ea-8e60-705b23f2938e.png)

Reply:
Also, the example is very long.  If you can shorten it or split it into multiple examples that would be better, also add some explanations along the way on the significant details that the reader should observe.  Really you should be ruthless about making examples as short and pointed as possible so that someone new to the feature can grasp them quickly.

And please start with `.. example:: INSERT TITLE FOR EXAMPLE`.

Your build didn't generate the reference manual, so I couldn't see the example with the coqtop processing.